internal/handler/userHandler_test.go

```diff
@@ -0,0 +1,290 @@
+package handler
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/gin-gonic/gin"
+	"github.com/google/uuid"
+	"github.com/murilogilfelpeto/ps-tag-onboarding-go/internal/handler/dto/request"
+	"github.com/murilogilfelpeto/ps-tag-onboarding-go/internal/handler/dto/response"
+	"github.com/murilogilfelpeto/ps-tag-onboarding-go/internal/handler/mapper"
+	"github.com/murilogilfelpeto/ps-tag-onboarding-go/internal/mocks"
+	"github.com/murilogilfelpeto/ps-tag-onboarding-go/internal/service/models"
+	"github.com/stretchr/testify/assert"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func TestSaveUserSuccess(t *testing.T) {
+	gin.SetMode(gin.TestMode)
+	recorder := httptest.NewRecorder()
+	ctx, _ := gin.CreateTestContext(recorder)
+	mockService := mocks.NewService(t)
+
+	requestBody := request.UserRequestDto{
+		FirstName: "John",
+		LastName:  "Doe",
+		Email:     "johndoe@email.com",
+		Age:       18,
+	}
+
+	jsonBody, _ := json.Marshal(requestBody)
+	ctx.Request = &http.Request{
+		Body: io.NopCloser(bytes.NewBuffer(jsonBody)),
+	}
+
+	user, _ := mapper.UserRequestToUser(requestBody)
+	createdUser, _ := models.NewUser("f7d2ea4b-a4d0-4103-9c63-55ec7977e4d1", "John", "Doe", "john.doe@email.com", 36)
+	mockService.EXPECT().SaveUser(ctx, user).Return(&createdUser, nil).Once()
+
+	handler := &Handler{
+		userService: mockService,
+	}
+
+	handler.Save(ctx)
+	var responseBody response.UserResponseDto
+	_ = json.Unmarshal(recorder.Body.Bytes(), &responseBody)
+
+	assert.Equal(t, http.StatusCreated, recorder.Code)
+	assert.Equal(t, createdUser.GetID(), responseBody.ID)
+	assert.Equal(t, createdUser.GetFirstName(), responseBody.FirstName)
+	assert.Equal(t, createdUser.GetLastName(), responseBody.LastName)
+	assert.Equal(t, createdUser.GetEmail(), responseBody.Email)
+	assert.Equal(t, createdUser.GetAge(), responseBody.Age)
+	mockService.AssertExpectations(t)
```

Comment:
This line is not required, when using .EXPECT() expectations are automatically checked when the test is run.